event/src/unvalidated/builder.rs

```diff
@@ -0,0 +1,277 @@
+    prev: Cid,
+    data: T,
+    controllers: Vec<String>,
+    sep: String,
+    additional: HashMap<String, Value>,
+}
+
+impl<T> Additional for DataBuilder<T> {
+    fn with_additional(mut self, key: String, value: Value) -> Self {
+        self.additional.insert(key, value);
+        self
+    }
+}
+
+impl<T> Controllers for DataBuilder<T> {
+    fn with_controller(mut self, controller: String) -> Self {
+        self.controllers.push(controller);
+        self
+    }
+}
+
+impl<T> Sep for DataBuilder<T> {
+    fn with_sep(mut self, sep: String) -> Self {
+        self.sep = sep;
+        self
+    }
+}
+
+impl<T> DataBuilder<T> {
+    /// Build data event
+    pub async fn build(self) -> anyhow::Result<DataPayload<T>> {
+        let header = DataHeader::new(self.controllers, self.additional);
+        let payload = DataPayload::new(self.id, self.prev, Some(header), self.data);
+        Ok(payload)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::unvalidated::payload::Payload;
+
+    const TEST_CID: &str = "bafyreiaroclcgqih242byss6pneufencrulmeex2ttfdzefst67agwq3im";
+
+    #[tokio::test]
+    async fn should_build_init_event_without_data() {
+        let event = Builder::default()
+            .with_sep("sep".to_string())
+            .with_controller("controller".to_string())
+            .init()
+            .build()
+            .await
+            .expect("failed to build event");
+        if let Payload::Init(event) = event {
+            assert_eq!(event.header.controllers, vec!["controller"]);
+            assert_eq!(event.header.sep, "sep");
+            assert!(event.data.is_none());
+        } else {
+            panic!("expected init event");
+        }
+    }
+
+    #[tokio::test]
+    async fn should_build_init_event_with_data() {
+        let event = Builder::default()
+            .init()
+            .with_sep("sep".to_string())
+            .with_controller("controller".to_string())
+            .with_data("data".to_string())
+            .build()
+            .await
+            .expect("failed to build event");
+        if let Payload::Init(event) = event {
+            assert_eq!(event.header.controllers, vec!["controller"]);
+            assert_eq!(event.header.sep, "sep");
+            assert_eq!(event.data, Some("data".to_string()));
+        } else {
+            panic!("expected unsigned event");
+        }
+    }
+
+    #[tokio::test]
+    async fn should_build_data_event() {
+        let event = Builder::default()
+            .data(
+                Cid::try_from(TEST_CID).unwrap(),
+                Cid::try_from(TEST_CID).unwrap(),
+                "data",
+            )
+            .with_controller("controller".to_string())
+            .with_sep("sep".to_string())
+            .with_additional("model".to_string(), "blah".to_string().into())
+            .build()
+            .await
+            .expect("failed to build event");
+        if let Payload::Data(event) = event {
```

Comment:
How do you feel about using `expect` test here as it make it quick and easy to validate that the entire event got constructed as expected?